CMakeLists.txt

```diff
@@ -5,6 +5,14 @@ if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
   message(FATAL_ERROR "In-source builds are prohibited.")
 endif ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
 
+# Import MFEM
+find_package(mfem REQUIRED NAMES MFEM)
+# Use the same C++ compiler as MFEM. This is needed when MFEM was built using
+# an MPI wrapper and we do not have explicitly the MPI compile and link flags.
+if (NOT CMAKE_CXX_COMPILER AND MFEM_CXX_COMPILER)
+  set(CMAKE_CXX_COMPILER "${MFEM_CXX_COMPILER}")
```

Comment:
This is actually just __before__ the `project()` command and works fine in all cases - with and without forcing an MPI wrapper (in MFEM) and with and without MPI. I know this is not the recommended way to configure MPI, but in some cases (e.g. some LLNL machines) it was the only way to get CMake to use the correct libraries.

Can you try this on your machines to see if it works?